src/lib/mdns/minimal/responders/IP.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "IP.h"
+
+#include <mdns/minimal/records/IP.h>
+
+namespace mdns {
+namespace Minimal {
+
+void IPv4Responder::AddAllResponses(const chip::Inet::IPPacketInfo * source, ResponderDelegate * delegate)
+{
+    for (chip::Inet::InterfaceAddressIterator it; it.HasCurrent(); it.Next())
+    {
+        if (it.GetInterfaceId() != source->Interface)
+        {
+            continue;
+        }
+
+        chip::Inet::IPAddress addr = it.GetAddress();
+        if (!addr.IsIPv4())
+        {
+            continue;
+        }
+        IPResourceRecord record(GetQName(), addr);
+        delegate->AddResponse(record);
+    }
+}
+
+void IPv6Responder::AddAllResponses(const chip::Inet::IPPacketInfo * source, ResponderDelegate * delegate)
+{
+    for (chip::Inet::InterfaceAddressIterator it; it.HasCurrent(); it.Next())
```

Comment:
This is a problem I think. This will list every possible IPv6 addresses on all interfaces, when in fact it should only respond the IPv6 addresses on the interface receiving the query.

See https://tools.ietf.org/html/rfc6762#section-14

```
   Except in the case of proxying and other similar specialized uses,
   addresses in IPv4 or IPv6 address records in Multicast DNS responses
   MUST be valid for use on the interface on which the response is being
   sent.
```

Reply:
The check right below is `if (it.GetInterfaceId() != source->Interface)` to filter out non-matching.